docs/docs/cmd/teams/channel/channel-add.md

```diff
@@ -22,6 +22,12 @@ m365 teams channel add [options]
 `-d, --description [description]`
 : The description of the channel to add
 
+`--type [type]`
+: Type of channel to create: `standard,private`. Default `standard`.
+
+`--owner [owner]`
+: User with this ID or UPN will be added as owner of the private channel. This option is required when type is `private`.
```

Comment:
Let's not make it required but add the currently logged-in user if no owner is specified. We should add this to the documentation too.

What do you think about this?

Reply:
I think it's a good idea. All I wonder is what we should do when people authenticate with an application token?

Reply:
Hi @milanholemans,
in that case, I would fallback to the behavior we have today. You could use the approach presented here: https://github.com/pnp/cli-microsoft365/pull/3088 to retrieve the information if we are in an application context or not< (method isAppOnlyAccessToken). I also implemented a method to retrieve the userId of the current user getUseridFromAccessToken (if you require it).

I haven't had time to implement these methods in the Utils.ts class yet. If you want, you can take these and reuse them in your logic?

Cheers

Reply:
All right, thanks for the tip!

Reply:
Hi @plamber
Is it possible that there is no accesstoken available when validating options? Right now I have following code in my `validate` function:
![image](https://user-images.githubusercontent.com/11723921/161608764-2e427a47-eb5b-4adb-9def-8c63a11e5cab.png)

However I keep getting this error:
![image](https://user-images.githubusercontent.com/11723921/161609076-c7418b6c-a1ad-4d59-9af4-d49d476c819d.png)

The provided code seems to be working correctly when executed in the `commandAction` function. Do you have any idea?

Reply:
Hi @milanholemans,
this could be indeed the case. @waldekmastykarz: do we have some other options to identify if we ar ein app or delegated scope when validating options?

If this is not possible, then @milanholemans, lets fallback to the previous functionality as you already implemented.

Thank you.

Reply:
The validation stage occurs before we restored auth so we don't have access to the access token in there. This type of validation we'd need to implement as part of the commandAction logic.

Reply:
So what should we do? Leave it as is, or should we move the option checks to the `commandAction` function?

Reply:
Just leave it as it is. Thank you

Reply:
We shouldn't throw an unhandled exception in our code, so I'd suggest that we do the necessary check in commandAction, especially as it's something that we know that could happen.

Reply:
> We shouldn't throw an unhandled exception in our code, so I'd suggest that we do the necessary check in commandAction, especially as it's something that we know that could happen.

Right now it's required to provide an `--owner` option when `--type private` is used. So according to me, it's not possible to get an unhandled exception neither in delegated context nor in application context.